dali/python/backend_impl.cc

```diff
@@ -1084,13 +1084,49 @@ PYBIND11_MODULE(backend_impl, m) {
         "show_tensors"_a = false,
         "show_ids"_a = false,
         "use_colors"_a = false)
-    .def("epoch_size", &Pipeline::EpochSize)
+    .def("epoch_size",  [](Pipeline* p, bool with_padding) {
+          return p->EpochSize(with_padding);
+        })
     .def("epoch_size",
-        [](Pipeline* p, const std::string& op_name) {
-          std::map<std::string, Index> sizes = p->EpochSize();
+        [](Pipeline* p, const std::string& op_name, bool with_padding) {
+          std::map<std::string, Index> sizes = p->EpochSize(with_padding);
           DALI_ENFORCE(sizes.find(op_name) != sizes.end(),
               "Operator " + op_name + " does not expose valid epoch size.");
           return sizes[op_name];
+        })
+    .def("shards_number", &Pipeline::NumberOfShards)
```

Comment:
Do we really want to expose those maps to Python in that way?

Reply:
This is the current API semantic. You can query about the EpochSize the same way. Wanted to be consistent.

Reply:
Done